### ISSUE TITLE
Feature Request: "Do not suspend this tab"

### ISSUE THREAD

**Issue by oucil** [Hau-Hau/firefox-tab-suspender#52]
I don't want to have to "Pin" the tabs that I want to keep open, there needs to be an option to temporarily add certain tabs to an ignore list.  Better yet, allow me to both temporarily (via context) add to the ignore list, but also manually maintain a permanent ignore list. I've lost SOOOO many half filled in forms as a result of them being discarded in the background while I research on other tabs.

Thanks!

**Comment by Adam-0-Moore** [comment]
This would be so great!

**Comment by davidhunter22** [comment]
I would agree. I have used the great suspender on chrome a lot a love it. I find without the whitelist of sites or tabs not to suspend the tab suspender is not usable. My ideal would be to have an item on the tab context menu allowing you to block suspension for some period of time say "1 min, 5 min, never clock this site"

**Comment by Hau-Hau** [comment]
Hello!

Thanks guys for idea!
I will check how it is solved in _The Great Suspender_!

Best regards!

**Comment by waqarrashid33** [comment]
Any timeline for this feature? I am having a hard time using this addon because I need to keep some tabs open but don't want to pin them.

**Comment by Hau-Hau** [comment]
Hi @waqarrashid33,
I am unable to indicate the time frame when any feature will be finished because I'm developing addon after work and studies, sometimes I'm just too tired to work on addon. 

But if anyone is interested in contributing  then any help will be great!

**Comment by aF4XevB4i1CC** [comment]
Just leaving my comment here, to also indicate my need for this feature! :-D

It will be a great add to the addon. I am having the same issue: If I do not want to loose data in forms or such, I either have to increase the timespan when a tab will be suspended to an amount, that the overall usage of the addon does not make sense anymore (e.g. 20 Minutes) or deactivate it. 

I agree with other commenters before me, that a managable whitelist would be great for this.

**Comment by Hau-Hau** [comment]
I'm don't sure how it should work.
Permanent white/black list is simple feature, but I don't know how should works temporary ignoring tabs. 

User is selecting 'Don't suspend this tab'  option in tab's context menu. 
Now tab will be ignored.
But how user should be informed about ignored tabs in this session? Other color of favicon, maybe small icon in left upper corner of favicon. Or maybe add to tab name "[Ignored]".

How in your opinion this feature should work?

Best regards!

**Comment by aF4XevB4i1CC** [comment]
I am not even sure, whether the user needs to "be informed" about a temporarily suspended tab? In comparison with the permanent white / black-list: would you think, that also those need a hint, that those are currently being not suspended? Of course it would be nice, if there was a UX-approach to it, that one "notices" it that a tab is being suspended or not (regardless if temporarily or permanent) ... but I, personally, would see that as a nice-to-have feature. As currently the tabs "indicate" their status (being currently suspended, or not) by slightly changing the color tone in the tab ... I would guess this is also enough for the black/whitelist ...

As a light-weight implementation, I would argue for it only being visible in some context menu, where the user sees a sub-menu list of the permanent white / black-list and in another menu a list with the temporarily suspended tabs.

Cheers

**Comment by oucil** [comment]
I agree with @aF4XevB4i1CC, I'm not concerned with being visually informed which are set to hibernate, vs those that are not, beyond the right click status, I don't like too many add-ons all trying to do their own thing to the UI, it gets very cluttered. That said, I'm sure many others would be interested in at least something, or the option of something.  My opinion would be to only visualize the ones that have been excluded from the standard behaviour.  Possibly a little "pause" icon in the bottom right of the favicon.  It's simple, not overly intrusive, and would only apply to tabs that you have specifically told not to hibernate.

**Comment by Hau-Hau** [comment]
Thanks guys, now I know everything what I need!

**Comment by joaoeudes7** [comment]
I'm looking forward to this functionality!

**Comment by WayneLambert** [comment]
This is a great extension and I agree with the others here that this would be a welcomed new feature. I imagine it as being something like _"Add to whitelist"_ on the context menu.

**Comment by khawajamechatronics** [comment]
I'm looking forward to this functionality!

**Comment by RJVB** [comment]
Adding my vote to this, as well as to an option not to suspend tabs with unsubmitted forms.

FWIW, I've been able at some point in the past to make myself a Firefox version of "The Great Suspender" via the "Chrome Store Foxified" extension. I'm not certain if that's still possible but my TGS version still works (mostly; sometimes tabs lose their capability to unsuspend but I've always been able to get back the page since the original URL is still in the `moz-extension://` incantation.

**Comment by atrifix** [comment]
There are two features I would like to see:
1. Permanently whitelist domain/URL/pattern
2. Temporarily whitelist all tabs in the current window (via context menu). Tabs in other windows will still be suspended. Resets upon closing the window.
